lib/buffer.js

```diff
@@ -105,6 +105,10 @@ function fromString(string, encoding) {
     encoding = 'utf8';
 
   var length = byteLength(string, encoding);
+
+  if (length === 0)
+    return new Buffer(0);
```

Comment:
@silverwind The code works fine, but we're allocating an unnecessary object. Though I doubt that has an impact on performance either. Part of the point is that Buffer is not an actual constructor anymore. It's now more of a factory method. But again it's nothing serious or is a change that needs to be made.